pkg/apis/serving/v1alpha1/service_validation.go

```diff
@@ -53,6 +73,32 @@ func (pt *PinnedType) Validate() *apis.FieldError {
 	return errs.Also(pt.Configuration.Validate().ViaField("configuration"))
 }
 
+// Validate validates the fields beloning to RunLatestType
 func (rlt *RunLatestType) Validate() *apis.FieldError {
 	return rlt.Configuration.Validate().ViaField("configuration")
 }
+
+// Validate validates the fields beloning to ManualType
+func (m *ManualType) Validate() *apis.FieldError {
+	return nil
+}
+
+// Validate validates the fields beloning to ReleaseType
+func (rt *ReleaseType) Validate() *apis.FieldError {
+	var errs *apis.FieldError
+
+	numRevisions := len(rt.Revisions)
+	if numRevisions < 1 || numRevisions > 2 {
+		errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("%v", numRevisions), "revisions.length"))
```

Comment:
Made <1 a missing field error.

For the >2 check none of the error messages felt quite right. Invalid value seemed the closest, and I think that leaving the path as "revisions" seems a bit confusing since the length is not the value of the field.

Some alternatives. Let me know if you prefer one of the below (or have a third option):
1. Introduce new error type specifically for length
2. Format the num revisions value as "length:%v" or "len(%v)"

Reply:
I think you can also synthesize an `apis.FieldError{}` directly :)

Reply:
Creating FieldError directly. Using 'expected, got' style used in other field errors.